fili-core/src/main/java/com/yahoo/bard/webservice/async/preresponses/stores/HashPreResponseStore.java

```diff
@@ -8,32 +8,74 @@
 
 import java.util.LinkedHashMap;
 import java.util.Map;
+import java.util.concurrent.CountDownLatch;
 
 /**
  * An in-memory implementation of PreResponseStore mainly for testing purposes. It only provides functionality to save
  * an entry to store and get an entry from the store. It does not have delete functionality nor does it take care of
  * cleaning stale data.
+ * <p>
+ * Since the HashPreResponseStore is intended primarily for testing, it also includes two maps of tickets to
```

Comment:
Some thoughts here:

1.) If this is primarily for testing, we should move this class to the `test` source root
2.) If this class _is_ useful for non-testing, then we should leave it here and remove the latches (assuming they are not useful for non-testing workloads)
3.) If we want _both_ (ie. keep this class for real use, and also give it the latch counting behavior for tests to use, then we should have a `LatchedHashPreResponseStore` in the `test` source root that extends this one, adding the latching behavior.

Either way, things that are only for testing don't belong in code that is in the `src` root. That's what the `test` root is for.